src/main/java/com/michelin/ns4kafka/service/executor/UserAsyncExecutor.java

```diff
@@ -200,6 +233,66 @@ public String resetPassword(String user) {
             return password;
         }
 
+        @Override
+        public void setPassword(String user, String password) {
```

Comment:
See suggestions from above:

➡️ I think `userAsyncExecutor#resetPassword` and `userAsyncExecutor#setPassword` can be a single method as well:

- Add a password parameter to userAsyncExecutor#resetPassword
- Extract the generation of the random password outside of `userAsyncExecutor#resetPassword` in order to generate it if the @Body is empty